pallets/asset/src/lib.rs

```diff
@@ -1350,6 +1315,25 @@ impl<T: Trait> Module<T> {
         T::MaxNumberOfTMExtensionForAsset::get()
     }
 
+    fn ensure_pia_or_owner_with_custody(
+        origin: T::Origin,
+        ticker: Ticker,
+    ) -> Result<PermissionedCallOriginData<T::AccountId>, DispatchError> {
+        let data = Identity::<T>::ensure_origin_call_permissions(origin)?;
+        let skey = data.secondary_key.as_ref();
+
+        // Ensure that the secondary key has asset permission
+        Self::ensure_asset_perms(skey, &ticker)?;
+
+        // Ensure that the sender is the PIA or the token owner and returns the PIA address.
+        let pia = Self::ensure_pia_or_owner(&ticker, data.primary_did)?;
+
+        // Ensure that the caller has relevant portfolio permissions
+        let portfolio = PortfolioId::default_portfolio(pia);
```

Comment:
I think here it get assumed that PIA and owner will always be the same but it is not always true. here owner did should be used.

Reply:
It's assumed that `data.primary_did == pia` always holds, which is the case:
```rust
    pub fn ensure_pia_or_owner(
        ticker: &Ticker,
        did: IdentityId,
    ) -> Result<IdentityId, DispatchError> {
        Self::token_details(&ticker)
            .primary_issuance_agent
            .filter(|pia| *pia == did)
            .or_else(|| Self::is_owner(&ticker, did).then(|| did))
            .ok_or_else(|| Error::<T>::Unauthorized.into())
    }
```